### PR TITLE
chore: aria label to aria labeledby

### PR DESCRIPTION
**Pull request by JasonLantz**
react modal released an api for aria a couple days ago....  this replaces the last implementation that used aria-label.

### CODE REVIEW

**Comment by umahaea** [comment]
Regression passed, so no problem with breaking changes :), let me know if there is any test that I can add in.

File: demo/translations/fr.json [hunk 1]
```diff
@@ -6,5 +6,6 @@
   "modalCancelButtonText" : "le Cancel button",
   "headerTitle"           : "French Basic Title",
   "closeButtonSRText"     : "french close dialog",
-  "bodyText"              : "Pass in JSX elements as children to the modal, You must pass in the text object with valid strings"
+  "bodyText"              : "Pass in JSX elements as children to the modal, You must pass in the text object with valid strings",
+  "srHeaderText"          : "French Hi im a modal"
```
Comment (StommePoes) [inline demo/translations/fr.json]: Hi je suis le modal

haw haw!

baguettes

**Comment by StommePoes** [comment]
Looks good.

Another option would have been to still require the modal-header text, and if there's a header then it just goes in the header, and if not would change aria-labelledby to aria-label="that modal-header string".

However switching attributes on the fly is possibly more painful than just having some sr-only text.

File: demo/translations/fr.json (hunk 1 above)
Comment (JasonLantz) [inline demo/translations/fr.json]: Bigger!

**Comment by JasonLantz** [comment]
the text is kinda hidden in a text object... i figured I can use react to validate and leaving it out will throw errors...

**Pull request merged**